src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -625,102 +610,73 @@ namespace xclhwemhal2 {
     if (xclemulation::config::getInstance()->isSharedFmodel() && !mVersalPlatform) {
       setenv("SDX_USE_SHARED_MEMORY","true",true);
     }
-    // iterate cores
-    count = 0;
-    for (auto& xml_core : xml_project.get_child("project.platform.device"))
-    {
-      if (xml_core.first != "core")
-        continue;
-      if (++count > 1)
-      {
-        //Give error and return from here
-      }
-    }
 
-    std::vector<std::string> kernels;
-
-    // iterate kernels
-    for (auto& xml_kernel : xml_project.get_child("project.platform.device.core"))
+    auto xclbin_object = new xrt::xclbin(args.m_top);
```

Comment:
This is a bug.  You can't construct on heap and not delete the heap memory when done.  This should not be on the heap to begin with.   Also, this is where some restructuring is called for, because creating the xclbin object shouldn't really be necessary given that xrt_core::device ends up containing / constructing this object also.   That's for a complete redesign, but for now consider moving the xrt::xclbin construction earlier so you can use it for xsa name and if xsa name isn't sufficient and you indeed need the fpgaDevice attribute value, then get that from the xrt::xclbin object through some internal (common/api/xclbin_int.h) method.